packages/plugin-google-analytics/lib/index.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-unused-vars, class-methods-use-this */
+import {
+  Event, Properties, RequestLoggerPlugin, PluginLoadOptions, PageOptions, IdentifyOptions, TrackOptions,
+} from '@itly/sdk';
+
+type GoogleAnalyticsOptions = {
+  gtag?: Gtag.Gtag;
+  measurementId: string;
+}
+
+export class GoogleAnalyticsPlugin extends RequestLoggerPlugin {
+  private _gtag?: Gtag.Gtag;
+
+  get gtag(): Gtag.Gtag {
+    this._gtag = this._gtag || self.gtag;
+    if (!this._gtag) throw new Error('No gtag provided to constructor, or available on window');
+    return this._gtag;
+  }
+
+  constructor(private opts: GoogleAnalyticsOptions) {
+    super('google-analytics');
+  }
+
+  load(options: PluginLoadOptions) {
+    super.load(options);
+    if (this.opts.gtag) {
+      this._gtag = this.opts.gtag;
+    } else if ('gtag' in self) {
+      this._gtag = self.gtag;
+    } else {
+      this.initializeGtag(this.opts.measurementId);
+    }
+  }
+
+  identify(userId: string | undefined, properties?: Properties | undefined) {
+    this.gtag('config', this.opts.measurementId, { userId, ...properties });
+  }
+
+  track(userId: string | undefined, { name, properties }: Event, options?: TrackOptions) {
```

Comment:
`options` here should be a `PluginCallOptions` subclass. [The core sdk now filters the top-level CallOptions by plugin id](https://github.com/iterativelyhq/itly-sdk/blob/master/packages/sdk/lib/base.ts#L344) so each plugin will only receive its own options object. E.g. [plugin-amplitude-node](https://github.com/iterativelyhq/itly-sdk/blob/master/packages/plugin-amplitude-node/lib/index.ts)